pkg/api/validate/dynamic/quota.go

```diff
@@ -11,115 +11,59 @@ import (
-		requiredResources["standardESv3Family"] += (count * 16)
-		requiredResources["cores"] += (count * 16)
-	case api.VMSizeStandardE32sV3:
-		requiredResources["standardESv3Family"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-		//Support for Compute isolation
-	case api.VMSizeStandardE64iV3:
-		requiredResources["standardEIv3Family"] += (count * 64)
-		requiredResources["cores"] += (count * 64)
-	case api.VMSizeStandardE64isV3:
-		requiredResources["standardEISv3Family"] += (count * 64)
-		requiredResources["cores"] += (count * 64)
-	case api.VMSizeStandardF4sV2:
-		requiredResources["standardFSv2Family"] += (count * 4)
-		requiredResources["cores"] += (count * 4)
-	case api.VMSizeStandardF8sV2:
-		requiredResources["standardFSv2Family"] += (count * 8)
-		requiredResources["cores"] += (count * 8)
-	case api.VMSizeStandardF16sV2:
-		requiredResources["standardFSv2Family"] += (count * 16)
-		requiredResources["cores"] += (count * 16)
-	case api.VMSizeStandardF32sV2:
-		requiredResources["standardFSv2Family"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-	case api.VMSizeStandardF72sV2:
-		requiredResources["standardFSv2Family"] += (count * 72)
-		requiredResources["cores"] += (count * 72)
-	case api.VMSizeStandardM128ms:
-		requiredResources["standardMSFamily"] += (count * 128)
-		requiredResources["cores"] += (count * 128)
-	case api.VMSizeStandardG5:
-		requiredResources["standardGFamily"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-	case api.VMSizeStandardGS5:
-		requiredResources["standardGSFamily"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-	case api.VMSizeStandardL4s:
-		requiredResources["standardLsFamily"] += (count * 4)
-		requiredResources["cores"] += (count * 4)
-	case api.VMSizeStandardL8s:
-		requiredResources["standardLsFamily"] += (count * 8)
-		requiredResources["cores"] += (count * 8)
-	case api.VMSizeStandardL16s:
-		requiredResources["standardLsFamily"] += (count * 16)
-		requiredResources["cores"] += (count * 16)
-	case api.VMSizeStandardL32s:
-		requiredResources["standardLsFamily"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-	case api.VMSizeStandardL8sV2:
-		requiredResources["standardLsv2Family"] += (count * 8)
-		requiredResources["cores"] += (count * 8)
-	case api.VMSizeStandardL16sV2:
-		requiredResources["standardLsv2Family"] += (count * 16)
-		requiredResources["cores"] += (count * 16)
-	case api.VMSizeStandardL32sV2:
-		requiredResources["standardLsv2Family"] += (count * 32)
-		requiredResources["cores"] += (count * 32)
-	case api.VMSizeStandardL48sV2:
-		requiredResources["standardLsv2Family"] += (count * 48)
-		requiredResources["cores"] += (count * 48)
-	case api.VMSizeStandardL64sV2:
-		requiredResources["standardLsv2Family"] += (count * 64)
-		requiredResources["cores"] += (count * 64)
-	default:
-		//will only happen if pkg/api verification allows new VMSizes
-		return fmt.Errorf("unexpected node VMSize %s", vmSize)
```

Comment:
> where is the source of truth for the vms we do support ? Is it the list of const in the api package ?

[Here](https://github.com/Azure/ARO-RP/blob/5668a316afae97b690871edac2c59eb8f84efacb/pkg/api/validate/vm.go#L14-L72) is the API version agnostic validation. I would say this is source of truth: if a VM size you specify is missing from here - you won't be able to create a cluster using this vm size no mater what API version you are using.

We also have lists of constants:
* In our internal representation [in `pkg/api`](https://github.com/Azure/ARO-RP/blob/5668a316afae97b690871edac2c59eb8f84efacb/pkg/api/openshiftcluster.go#L274-L313) which should cointain all the supported VMs as well.
* In each external representation/API version in `pkg/api/v*` pacakges. These lists can be shorter. [Example](https://github.com/Azure/ARO-RP/blob/5668a316afae97b690871edac2c59eb8f84efacb/pkg/api/v20200430/openshiftcluster.go#L134-L156) of a shorter list.

> Also the original function had the same kind of checking anyways, in the default case.

I'm sorry, I think I made a wrong suggestion here. We do need this check. I still think that an unit test can be useful here and will help us to catch this issue at testing time instead of runtime. But this is out of scope for your refactoring. If you like, you can look into adding an unit test in a separate PR ;)